### PR TITLE
mlperf enwiki conversion code mild cleanup

### PR DESCRIPTION
**Pull request by knighton**

### CODE REVIEW

File: streaming/text/convert/enwiki/mds/make_train_parallel.py [hunk 1]
```diff
@@ -1,28 +1,68 @@
...
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Returns:
+        Namespace: Command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--in_pattern',
+                      type=str,
+                      default='/tmp/enwiki_preproc/results4/part-%05d-of-00500')
```
Comment (karan6181) [inline streaming/text/convert/enwiki/mds/make_train_parallel.py:19]: Is the `default` will always exist? Wondering do we need `default` or can we make it mandatory ?

File: streaming/text/convert/enwiki/mds/make_train_parallel.py [hunk 2]
```diff
@@ -1,28 +1,68 @@
...
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Returns:
+        Namespace: Command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--in_pattern',
```
Comment (karan6181) [inline streaming/text/convert/enwiki/mds/make_train_parallel.py:17]: Can you please add `help` for all the arguments to understand what each argument means? Thanks!

**Review by dskhudia — APPROVED** [review]
__getitem changes look fine to me.

File: streaming/text/convert/enwiki/mds/merge_shard_groups.py [hunk 3]
```diff
@@ -1,40 +1,72 @@
...
+        shards_this_group = len(os.listdir(subdir)) - 1
+
+        # Move shard files.
+        for shard in range(shards_this_group):
+            old_filename = f'{subdir}/shard.{shard:05d}.mds.zstd'
+            new_filename = f'{args.out_root}/shard.{offset + shard:05d}.mds.zstd'
+            os.rename(old_filename, new_filename)
+
+        # Collect shard infos.
+        index_filename = f'{subdir}/index.json'
```
[the remark this answers is not in the document]
Comment (knighton) [reply streaming/text/convert/enwiki/mds/merge_shard_groups.py:46]: Done

File: streaming/text/convert/enwiki/mds/merge_shard_groups.py [hunk 4]
```diff
@@ -1,40 +1,72 @@
...
+    for shard, info in enumerate(infos):
+        info['raw_data']['basename'] = f'shard.{shard:05d}.mds'
+        info['zip_data']['basename'] = f'shard.{shard:05d}.mds.zstd'
+
+    # Create new index.
+    obj = {
+        'version': 2,
+        'shards': infos,
+    }
+    index_filename = f'{args.out_root}/index.json'
```
[the remark this answers is not in the document]
Comment (knighton) [reply streaming/text/convert/enwiki/mds/merge_shard_groups.py:63]: Done

File: streaming/text/convert/enwiki/mds/make_train_parallel.py (hunk 1 above)
Comment (knighton) [reply streaming/text/convert/enwiki/mds/make_train_parallel.py]: Done

File: streaming/text/convert/enwiki/mds/merge_shard_groups.py [hunk 5]
```diff
@@ -1,40 +1,72 @@
...
+
+
+def parse_args() -> Namespace:
+    """Parse commmand-line arguments.
+
+    Returns:
+        Namespace: Command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--in_root', type=str, default='/tmp/mds-enwiki/train/')
```
[the remark this answers is not in the document]
Comment (knighton) [reply streaming/text/convert/enwiki/mds/merge_shard_groups.py]: Done

File: streaming/text/convert/enwiki/mds/make_train_parallel.py (hunk 2 above)
Comment (knighton) [reply streaming/text/convert/enwiki/mds/make_train_parallel.py:21]: Done

File: streaming/text/convert/enwiki/mds/README.md [hunk 6]
```diff
@@ -1,12 +1,17 @@
 ## English Wikipedia MDS dataset creation
 
-### Training split:
+### Setup:
+
+- Install tensorflow: `pip3 install tensorflow`
+- Download `results_text.zip`: [Google Drive](https://drive.google.com/drive/folders/1cywmDnAsrP5-2vsr8GDc6QUc7VWe-M3v)
```
[the remark this answers is not in the document]
Comment (knighton) [reply streaming/text/convert/enwiki/mds/README.md:6]: Punting on this cc: @bandish-shah

**Pull request merged**